### PR TITLE
test(e2e): Refactor field tests to use common functions

### PR DESCRIPTION
**Pull request by jstockwin**
cc @webteckie @joerter 

Following our discussions on slack, and for work on #2291. See what you think. 

The functions that are run by every test are now in a common file, `commonFieldTestUtils`, and all the `uiTest....Field` and `uxTest...Field` stuff imports these functions. Saves on repeated code (hence the 1,245 deletions compared to 253 additions).

### CODE REVIEW

File: test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js [hunk 1]
```diff
@@ -1,46 +1,8 @@
...
-	// UNDO ANY STATE CHANGES -- THIS TEST SHOULD RUN LAST
-	'restoring test state': function (browser) {
-		browser.initialFormPage.section.form
-			.click('@cancelButton');
-
-		browser.app
-			.waitForElementVisible('@listScreen');
-	},
+module.exports = {
+	before: fieldTests.beforeUI,
```
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js]: why `beforeUI`?

Comment (jstockwin) [inline test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js]: Because I had it in my head that the before functions were different for UI and UX, since using the UI before in the UX tests didn't work when I tried it. But yeah, there's no difference apart from the additional `browser`...

File: test/e2e/adminUI/tests/group005Fields/code/uxTestCodeField.js [hunk 2]
```diff
@@ -1,80 +1,8 @@
...
-			.verifyInput({value: 'Some Test Code for Field B'});
-	},
-	// UNDO ANY STATE CHANGES -- THIS TEST SHOULD RUN LAST
-	'restoring test state': function (browser) {
-	},
+module.exports = {
+	before: fieldTests.beforeUX,
+	after: fieldTests.after,
+	'Code field can be filled via the initial modal': fieldTests.assertInitialFormUX('Code', {value: 'Some test code for field A'}),
+	'Code field can be filled via the edit form': fieldTests.assertEditFormUX('Code', {value: 'Some test code for field B'}),
```
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/code/uxTestCodeField.js]: in cases like this, what if we want to target the input value to something other than field B?

File: test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js (hunk 1 above)
Comment (jstockwin) [inline test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js]: @webteckie Calling `fieldTests.beforeUI` in place of `fieldTests.beforeUX` for `url` results in the following: 
```
Running:  Url field can be filled via the initial modal
? TypeError: Cannot read property 'expect' of undefined
    at Object.Url field can be filled via the initial modal (D:\Documents\Development\GitHub\keystone\test\e2e\adminUI\tests\group005Fields\commonFieldTestUtils.js:88:5)
    at Module.call (D:\Documents\Development\GitHub\keystone\node_modules\nightwatch\lib\runner\module.js:63:34)
    at D:\Documents\Development\GitHub\keystone\node_modules\nightwatch\lib\runner\testcase.js:97:29
    at _fulfilled (D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:834:54)
    at self.promiseDispatch.done (D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:863:30)
    at Promise.promise.promiseDispatch (D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:796:13)
    at D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:556:49
    at runSingle (D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:137:13)
    at flush (D:\Documents\Development\GitHub\keystone\node_modules\q\q.js:125:13)
    at _combinedTickCallback (node.js:370:9)
    at process._tickCallback (node.js:401:11)

FAILED:  1 assertions failed (17ms)
```

File: test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		browser.listPage = browser.page.list();
+		browser.initialFormPage = browser.page.initialForm();
+
+		browser.app.navigate();
+		browser.app.waitForElementVisible('@signinScreen');
+
+		browser.signinPage.signin();
+		browser.app.waitForElementVisible('@homeScreen');
+	},
+	beforeUX: function (browser) {
```
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js]: I would consolidate beforeUI and beforeUX..I don't think it hurts anything to have the item page included in both cases.

File: test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+			browser.listPage
+				.click('@createFirstItemButton');
+
+			browser.app
+				.waitForElementVisible('@initialFormScreen');
+
+			browser.initialFormPage.section.form.section[ nameLowercase + 'List' ].section.name
+				.verifyUI();
+
+			browser.initialFormPage.section.form.section[ nameLowercase + 'List' ].section.fieldA
```
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js]: why is this line here?

Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js]: ah, never mind :-)

Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/commonFieldTestUtils.js]: it's confusing that the function parameter is called `name` and section. `name`  perhaps call the input parameter `listName`?

File: test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js (hunk 1 above)
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js]: yeah, you need the item page there...so just rename `beforeUX` to  `before` and use that and you should be good

Comment (jstockwin) [inline test/e2e/adminUI/tests/group005Fields/code/uiTestCodeField.js]: Okay, I understand what the issue was now. Will push a fix shortly.

File: test/e2e/adminUI/tests/group005Fields/code/uxTestCodeField.js (hunk 2 above)
Comment (webteckie) [inline test/e2e/adminUI/tests/group005Fields/code/uxTestCodeField.js]: perhaps something like this:
```
'Code field can be filled via the edit form': fieldTests.assertEditFormUX('Code', {
    fieldB: {
                  value: 'Some test code for field B'
              }
 }),
```

**Comment by jstockwin** [comment]
Closing in favour of #2736
